tests/io/test_spotter.py

```diff
@@ -32,3 +34,30 @@ def test_stats(self, wavespectra_name, spotter_name, kwargs):
         wavespectra = getattr(self.dset.spec, wavespectra_name)(**kwargs).values
         spotter = getattr(self.spot, spotter_name)
         assert wavespectra == pytest.approx(spotter, rel=1e-1)
+
+
+# class TestSpotterCsv:
+#     """Test parameters Spotter CSV reader."""
+
+#     @classmethod
+#     def setup_class(self):
+#         """Setup class."""
+#         self.dset = read_spotter_csv(os.path.join(FILES_DIR,'spotter_20210929.csv'))
+
+#     @pytest.mark.parametrize(
+#         "wavespectra_name, spotter_name, kwargs",
+#         [
+#             ("hs", "significantWaveHeight", {}),
+#             ("tp", "peakPeriod", {"smooth": False}),
+#             ("tm01", "meanPeriod", {}),
+#         ],
+#     )
+#     def test_stats(self, wavespectra_name, spotter_name, kwargs):
+#         """Assert that stat calculated from wavespectra matches the one read from spotter."""
+#         wavespectra = getattr(self.dset.spec, wavespectra_name)(**kwargs).values
```

Comment:
I've never used the class-based pytest structure you're using here and I'm also having a hard time understanding these lines (I think that you are comparing bulk parameters reported in the Spotter file to those you calculate from the spectral density, but I don't quite get how this works, particularly where in the `read_spotter` `JSON` workflow the `.dset.spec` field is created): https://github.com/wavespectra/wavespectra/blob/1a5a02fbbea3ab332e8b9b14d44e90beac631f54/tests/io/test_spotter.py#L32-L33

Reply:
These tests use this `@pytest.mark.parametrize` to allow running the function that comes right after (`test_stats`) with a range of input argument combinations (which are defined in that list of tuples in the arg of parametrize  decorator).

`dset.spec` is not actually created by the json reader per se - but it is the `SpecArray` accessor that wavespectra itself defines when it is imported and it finds a dataset with appropriate conventions. You don't have to worry about it, the important thing to know is that you have to return the xarray dataset from your reader with the appropriate conventions (specified [here](https://wavespectra.readthedocs.io/en/latest/conventions.html)).

You can define your tests in a different way that makes sense to you, as long as the tests are robust and accesses as much as the functionality in your code as possible in order to test the code well. But you got the existing one basically right, it simply compares stats calculated from wavespectra against those defined in the json file.

Reply:
Oh cool! While I don't really understand how his happens (something [here](https://github.com/wavespectra/wavespectra/blob/master/wavespectra/specdataset.py#L37) I think), I think I understand how to interact with it now.